.github/workflows/ci.yml

```
@@ -0,0 +1,58 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: CI Build
+
+# Controls when the workflow will run.
+on:
+  # run pipeline on push event of main branch
+  push:
+    branches:
+      - main
+  # run pipeline on pull request
+  pull_request:
+  # run pipeline on merge queue
+  merge_group:
+  # run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      create_release:
+        type: boolean
+        description: Create a (pre-)release when CI passes
+        default: false
+        required: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-ci-build
+  cancel-in-progress: true
+
+jobs:
+  build-test-release:
+    name: build-test-release
+    uses: "./.github/workflows/build-test-release.yml"
+    permissions:
+      contents: write
+    with: 
+      # create_release becomes true if the event that triggered this workflow is "workflow_dispatch" and inputs.create_release is true
+      # create_release becomes true if the event that triggered this workflow is "push" on main
+      # otherwise create_release becomes false
+      create_release: ${{ github.event_name == 'workflow_dispatch' && inputs.create_release || github.event_name == 'push' && true || false }}
```

please add braces for readability
```suggestion
      create_release: ${{ (github.event_name == 'workflow_dispatch' && inputs.create_release) || (github.event_name == 'push' && true) || false }}
```

also, what's the reason for the `&& true` and `|| false`?

---

I think they are redundant, removed them. I will be experimenting with the shorter version on another branch, will let you know the results